File: src/main/java/se/lu/nateko/cp/data/formats/bintable/BinTableFile.java
package se.lu.nateko.cp.data.formats.bintable;

import java.io.Closeable;
import java.io.File;
import java.io.FileNotFoundException;
import java.io.RandomAccessFile;

public abstract class BinTableFile implements Closeable{

	protected final RandomAccessFile file;
	protected final String fileName;
	protected final Schema schema;
	protected final long[] columnSizes;
	protected final long[] columnOffsets;
	protected final int[] columnValueSizes;

	protected BinTableFile(File file, Schema schema, String accessType) throws FileNotFoundException{
		this.file = new RandomAccessFile(file, accessType);
		this.fileName = file.getName();
		this.schema = schema;
		columnSizes = Utils.getColumnSizes(schema);
		columnOffsets = Utils.getColumnByteOffsets(columnSizes);
		columnValueSizes = Utils.getColumnValueSizes(schema);
	}
	
}

File: src/main/scala/se/lu/nateko/cp/data/formats/bintable/BinTableFile.scala
package se.lu.nateko.cp.data.formats.bintable

import java.io.Closeable
import java.io.File
import java.io.RandomAccessFile

trait BinTableFile extends Closeable:
	val schema: Schema

	def file(file: File, accessType: String) = new RandomAccessFile(file, accessType)

	protected val columnSizes = schema.columns.map(col => schema.size * col.byteSize)
	protected val columnOffsets = columnSizes.scan(0L)(_ + _)
	protected val columnValueSizes = schema.columns.map(_.byteSize)

